src/Backend/Modules/Mailmotor/Domain/Settings/Command/SaveSettingsHandler.php

```diff
@@ -26,24 +27,56 @@ public function __construct(
 
     public function handle(SaveSettings $settings): void
     {
-        // Define module
-        $module = 'Mailmotor';
-
-        // set our settings
-        $this->modulesSettings->set($module, 'mail_engine', $settings->mailEngine);
-        $this->modulesSettings->set($module, 'double_opt_in', $settings->doubleOptIn);
-        $this->modulesSettings->set($module, 'overwrite_interests', $settings->overwriteInterests);
-        $this->modulesSettings->set($module, 'automatically_subscribe_from_form_builder_submitted_form', $settings->automaticallySubscribeFromFormBuilderSubmittedForm);
+        $this->saveSetting('mail_engine', $settings->mailEngine);
+        $this->saveSetting('double_opt_in', $settings->doubleOptIn);
+        $this->saveSetting('overwrite_interests', $settings->overwriteInterests);
+        $this->saveSetting(
+            'automatically_subscribe_from_form_builder_submitted_form',
+            $settings->automaticallySubscribeFromFormBuilderSubmittedForm
+        );
 
         // mail engine is empty
         if ($settings->mailEngine === 'not_implemented') {
-            $this->modulesSettings->delete($module, 'api_key');
-            $this->modulesSettings->delete($module, 'list_id');
+            $this->deleteSetting('api_key');
+            $this->deleteSetting('list_id');
+
+            foreach (Language::getActiveLanguages() as $language) {
+                $this->deleteSetting('list_id_' . $language);
+            }
 
             return;
         }
 
-        $this->modulesSettings->set($module, 'api_key', $settings->apiKey);
-        $this->modulesSettings->set($module, 'list_id', $settings->listId);
+        $this->saveSetting('api_key', $settings->apiKey);
+        $this->saveSetting('list_id', $settings->listId);
+        $this->saveLanguageListIds($settings->languageListIds);
+    }
+
+    private function saveSetting(string $key, $value): void
```

Comment:
Isn't this bogus code? In my opinion this is not improving/simplifying the code, it is just "aliasing" a method.
I think it is wrong to alter the terminology we use. We `set` a setting, not `save`.

Reply:
changed in 81acd5752